core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java

```diff
@@ -582,7 +602,9 @@ public void serviceAction(HttpServletRequest request, HttpServletResponse respon
             logConfigurationException(request, e);
             sendError(request, response, HttpServletResponse.SC_NOT_FOUND, e);
         } catch (Exception e) {
-            e.printStackTrace();
+            if (printStackTraceOnException) {
```

Comment:
Instead of using a dedicated flag I would reuse the existing one - `devMode` in such case. I think the my purpose of putting `e.printStackTrace()` here was to fix some issues with unit tests, not a production related one.

Reply:
And there is already an `if` statement below `if (handleException || devMode) {` LOG can go there.